File: Resources/Lecture Slides/Python vs C++/Sample Code/3 If Statements/main.py
# Python vs. C++ Example 3
# Rachel J. Morris, 2013 - RachelJMorris@gmail.com, Moosader.com, github.com/Moosader
# MIT License
 
import random

random.seed( 3 )

playerDead = False
monsterDead = False
playerIsBlocking = False

playerHP = 100
monsterHP = 100

# GAME LOOP - Game repeats until the Player or Monster die
while ( playerDead == False and monsterDead == False ):
	# Reset any commands from last time
	playerIsBlocking = False
	
	# Display menu
	print( "\n" )
	print( "------------------------" )
	print( "Your HP: \t" + str( playerHP ) )
	print( "Monster HP: \t" + str( monsterHP ) )
	print( "------------------------" )
	print( "A. Attack" )
	print( "B. Block" )
	print( "H. Heal" )
	
	# Ask the player for their choice
	choice = input( "What will you do? " ) 
	
	#ATTACK
	if ( choice == "a" or choice == "A" ):
		# Randomly generate attack damage
		playerDamage = random.randint( 0, 11 )
		print()
		print( "You swing your sword!" )
		print( "You hit the monster for " + str( playerDamage ) + " damage!" )
		
		# Remove damage from the monster's HP, check if it's dead
		monsterHP -= playerDamage
		if ( monsterHP <= 0 ):
			print( "You have slain the monster!" )
			monsterDead = True
		
	# BLOCK
	elif ( choice == "b" or choice == "B" ):
		print( "You block!" )
		playerIsBlocking = True
		
	# HEAL
	elif ( choice == "h" or choice == "H" ):
		print( "You use a potion!" )
		healAmount = random.randint( 0, 5 )
		print( "You heal " + str( healAmount ) + " HP!" )
		
		playerHP += healAmount
		if ( playerHP > 100 ):
			playerHP = 100
	
	# Monster's turn (if it's still alive)
	if ( monsterDead == False ):
		# Monster attacks! Randomly generate damage
		monsterDamage = random.randint( 0, 10 )
		
		if ( playerIsBlocking ):
			monsterDamage = monsterDamage / 2
		
		print()
		print( "The monster claws at you!" )
		print( "The monster does " + str( monsterDamage ) + " damage to you!" )
		
		# Remove damage from player's HP, check if you're dead
		playerHP -= monsterDamage
		if ( playerHP <= 0 ):
			print( "You have fallen!" )
			playerDead = True	
# Main game loop

if ( playerDead ):
	print( "\n\nMan, who hired you to hunt monsters? You suck." )

elif ( monsterDead ):
	print( "\n\nYou killed the monster and are awarded with fifty mansions!" )

File: Resources/Lecture Slides/Python vs C++/Sample Code/3 If Statements/main.cpp
/*
 * Python vs. C++ Example 3
 * Rachel J. Morris, 2013 - RachelJMorris@gmail.com, Moosader.com, github.com/Moosader
 * MIT License
*/

#include <iostream>
#include <stdlib.h>
#include <string>
using namespace std;

int main()
{
	// Seed the random number generator
	srand( 3 );
	
	bool playerDead = false;
	bool monsterDead = false;
	bool playerIsBlocking = false;
	
	int playerHP = 100;
	int monsterHP = 100;
	
	string choice;
	
	/* GAME LOOP - Game repeats until the Player or Monster die */
	while ( !playerDead && !monsterDead )
	{
		// Reset any commands from last time:
		playerIsBlocking = false;
		
		// Display the player menu
		cout << endl << endl;
		cout << "-----------------------------" << endl;
		cout << "Your HP: \t" << playerHP << endl;
		cout << "Monster HP: \t" << monsterHP << endl;
		cout << "-----------------------------" << endl;
		cout << "A. Attack" << endl;
		cout << "B. Block" << endl;
		cout << "H. Heal" << endl;
		
		// Ask the player for their choice
		cout << "What will you do?" << endl;
		cin >> choice;
		
		/* ATTACK */
		if ( choice == "a" || choice == "A" )
		{
			// Randomly generate attack damage
			int playerDamage = rand() % 11;
			cout << endl;
			cout << "You swing your sword!" << endl;
			cout << "You hit the monster for " << playerDamage << " damage!" << endl;
			
			// Remove damage from the monster's HP, check if it's dead
			monsterHP -= playerDamage;
			if ( monsterHP <= 0 )
			{
				cout << "You have slain the monster!" << endl;
				monsterDead = true;
			}
		}
		/* BLOCK */
		else if ( choice == "b" || choice == "B" )
		{
			cout << "You block!" << endl;
			playerIsBlocking = true;
		}
		/* HEAL */
		else if ( choice == "h" || choice == "H" )
		{
			cout << "You use a potion!" << endl;
			int healAmount = rand() % 5;
			cout << "You heal " << healAmount << " HP!" << endl;
			
			playerHP += healAmount;
			if ( playerHP > 100 )
			{
				playerHP = 100;
			}
		}
		
		/* MONSTER ATTACKS (if it's not dead) */
		if ( !monsterDead )
		{
			// Now monster attacks you!
			// Randomly generate attack damage
			int monsterDamage = rand() % 10;
			if ( playerIsBlocking )
			{
				monsterDamage = monsterDamage / 2;
			}
			
			cout << endl;
			cout << "The monster claws at you!" << endl;
			cout << "The monster does " << monsterDamage << " damage to you !" << endl;
			
			// Remove damage from player's HP, check if you're dead
			playerHP -= monsterDamage;			
			if ( playerHP <= 0 )
			{
				cout << "You have fallen!" << endl;
				playerDead = true;
			}
		}
	}
	
	if ( playerDead )
	{
		cout << "\n\nMan, who hired you to hunt monsters? You suck." << endl;
	}
	else if ( monsterDead )
	{
		cout << "\n\nYou killed the monster and are awarded with fifty mansions!" << endl;
	}
	
	return 0;
} 
